functions/go/create-setters/createsetters/create_setters.go

```diff
@@ -0,0 +1,342 @@
+	return object.VisitFields(func(node *yaml.MapNode) error {
+		if node == nil || node.Key.IsNil() || node.Value.IsNil() {
+			// don't do IsNilOrEmpty check cs empty sequences are allowed
+			return nil
+		}
+
+		// the aim of this method is to create-setter for sequence nodes
+		if node.Value.YNode().Kind != yaml.SequenceNode {
+			// return if it is not a sequence node
+			return nil
+		}		
+
+		// add the key to the field path
+		fieldPath := strings.TrimPrefix(fmt.Sprintf("%s.%s", path, node.Key.YNode().Value), ".")
+
+		elements, err := node.Value.Elements()
+		if err != nil {
+			return errors.Wrap(err)
+		}
+		// extracts the values in sequence node to an array
+		var nodeValues []string
+		for _, values := range elements {
+			nodeValues = append(nodeValues, values.YNode().Value)
+		}
+		sort.Strings(nodeValues)
+
+		// checks if any of the values of node matches with ScalarSetters
+		// changes the node to FoldedStyle
+		nodeToAddComment := node.Value
+		if nodeToAddComment.YNode().Style == yaml.FlowStyle {
+			if hasMatchValue(nodeValues, cs.ScalarSetters) {
+				// changing the node style to FoldedStyle
+				nodeToAddComment.YNode().Style = yaml.FoldedStyle
+				// To add the comment to the key for the FoldedStyle value node
+				nodeToAddComment = node.Key
+			}
+		}
+
+		for _, arraySetters := range cs.ArraySetters {
+			// checks if all the values in node are present in array setter
+			if checkEqual(nodeValues, arraySetters.Values) {
+				if nodeToAddComment.YNode().Style == yaml.FlowStyle && len(nodeValues) > 0{
+					nodeToAddComment.YNode().Style = yaml.FoldedStyle
+					nodeToAddComment = node.Key
+				}
+				nodeToAddComment.YNode().LineComment = fmt.Sprintf("kpt-set: ${%s}", arraySetters.Name)
+				return nil
+			}
+		}
+
+		cs.Results = append(cs.Results, &Result{
+			FilePath:  cs.filePath,
+			FieldPath: fieldPath,
+			Value:     fmt.Sprint(nodeValues),
+			Comment:   nodeToAddComment.YNode().LineComment,
+		})
+		return nil
+	})
+}
+
+/**
+visitScalar accepts the input scalar node and performs following steps,
+checks if it is a scalar node
+adds the linecomment if it's value matches with any of the setter
+
+e.g.for input of scalar node 'nginx:1.7.1' in the yaml node
+
+apiVersion: v1
+...
+  env: dev
+  image: nginx:1.7.1
+
+and for input CreateSetters [[name: image, value: nginx], [name: env, value: dev], [name: tag, value: 1.7.1]]
+The yaml node is transformed to
+
+apiVersion: v1
+...
+  env: dev # kpt-set: ${env}
+  image: nginx:1.7.1 # kpt-set: ${image}:${tag}
+
+*/
+
+func (cs *CreateSetters) visitScalar(object *yaml.RNode, path string) error {
+	if object.YNode().Kind != yaml.ScalarNode {
+		// return if it is not a scalar node
+		return nil
+	}
+
+	linecomment, valueMatch := getLineComment(object.YNode().Value, cs.ScalarSetters)
+
+	// sets the linecomment if the match is found
+	if valueMatch {
+		object.YNode().LineComment = fmt.Sprintf("kpt-set: %s", linecomment)
```

Comment:
you should add results only within this if condition.

Reply:
I Will update that.